cabinets/__init__.py

```diff
@@ -52,6 +52,11 @@ def from_uri(uri) -> (Cabinet, str):
     return cabinet_, path
 
 
+def set_configuration(protocol, **kwargs):
+    cabinet_cls = SUPPORTED_PROTOCOLS[protocol]
```

Comment:
Maybe `get` here and raise a `CabinetError` (or whatever custom exceptions we spun up) if it's `None`?